### PR TITLE
Feat/fsdp

### PR DESCRIPTION
**Pull request by Abhrant**
Adds FSDP and changes nyuntam and nyuntam_adapt versions

### CODE REVIEW

File: zero/core/utils.py [hunk 1]
```diff
@@ -255,7 +255,7 @@ def run_docker_container(
                 source=str(NyunServices),
                 target=str(DockerPath.NYUN_SERVICES.value),
                 type="bind",
-                read_only=True,
+                read_only=False,
```
Comment (shwu-nyunai) [inline zero/core/utils.py:258]: we must not make this path writable.

File: zero/core/utils.py [hunk 2]
```diff
@@ -275,7 +275,7 @@ def run_docker_container(
             command=command,
             image=str(image[0]),
             device_requests=device_requests,
-            detach=True,
+            detach=False,
```
Comment (shwu-nyunai) [inline zero/core/utils.py:278]: based on the offline discussions, I assume this is the way to have access to the running container for errors and other outputs??
1. if so, can u elaborate on this a little? 
2. also, how are we now handling the outputs from the container? does it automatically return errors and raise exceptions? 
3. if so, how does the cli respond to it?
- lets confirm that the cli Aborts as expected when the docker raises exceptions or returns failures.
- the outputs from the docker are well casted and viewed on the terminal. (catching exceptions, outputs, converting bstrings to correct strings for outputs, etc.

File: zero/core/utils.py [hunk 3]
```diff
@@ -343,10 +343,9 @@ def get_service_from_metadata_extension_type(extension_type: WorkspaceExtension)
     if extension_type in {
         WorkspaceExtension.TEXT_GENERATION,
         WorkspaceExtension.VISION,
+        WorkspaceExtension.ADAPT,
```
Comment (shwu-nyunai) [inline zero/core/utils.py:346]: we updated the conditionals; lets also update the names of the keys, variables and imports accordingly

**Review by shwu-nyunai — CHANGES_REQUESTED** [review]

File: zero/core/utils.py (hunk 3 above)
Comment (Abhrant) [reply zero/core/utils.py:346]: what? 🤔

Comment (shwu-nyunai) [reply zero/core/utils.py:346]: the name of variables and imports; change them accordingly

Comment (Abhrant) [reply zero/core/utils.py:345]: done.

File: zero/core/utils.py (hunk 1 above)
Comment (Abhrant) [reply zero/core/utils.py]: done.

File: zero/core/utils.py (hunk 2 above)
Comment (Abhrant) [reply zero/core/utils.py:277]: > also, how are we now handling the outputs from the container? does it automatically return errors and raise exceptions?

It raises exceptions in the zero.log file 


> if so, how does the cli respond to it?

cli stops and shows aborted.

**Review by shwu-nyunai — CHANGES_REQUESTED** [review]
let's wait for the submodules to be updated. once done, we can add the version bumps to this pr itself and merge.

File: zero/core/utils.py (hunk 2 above)
Comment (Abhrant) [reply zero/core/utils.py:277]: I don't have any such references... removing it worked so I did it

Comment (Abhrant) [reply zero/core/utils.py:277]: Let's have this as an issue. For I can't find anything as of now. We can have it as an issue to keep it as an reminder.

File: zero/core/constants.py [hunk 4]
```diff
@@ -271,7 +271,8 @@ def get_service_path_in_docker(service_name: str):
 
 class DockerCommand(StrEnum):
 
-    RUN = "python main.py --yaml_path {script_path}"
+    RUN = "python run_dist.py --yaml_path {script_path}"
+    RUN_WITH_FSDP = ""
```
[the remark this answers is not in the document]
Comment (Abhrant) [reply zero/core/constants.py]: done.

File: zero/core/utils.py (hunk 2 above)
Comment (shwu-nyunai) [reply zero/core/utils.py:277]: sure, can u raise an issue and link it here? @Abhrant 
marking resolved for now.

File: zero/services/nyuntam
Comment (shwu-nyunai) [inline zero/services/nyuntam:1]: lets update the commit id with the latest main. there are many fixes added after this commit id.

**Review by shwu-nyunai — CHANGES_REQUESTED** [review]

**Review by Arnav0400 — APPROVED** [review]

File: zero/services/nyuntam
Comment (Arnav0400) [reply zero/services/nyuntam:1]: is this done?

**Review by shwu-nyunai — APPROVED** [review]
lgtm.

**Pull request merged**